testing/tests/test_e2e/test_secrets.py

```diff
@@ -661,17 +662,65 @@ def test_add_secret_with_metadata(secrets_context: Context[SecretsCharm], fields
         assert info['rotates'] is None
 
 
+@pytest.mark.parametrize('lookup_by', ['id', 'label'])
+@pytest.mark.parametrize(
+    'flow',
+    [
+        'content,label,description,expire,rotate',
+        'content,description,content,description',
+        'rotate,content,rotate,content,rotate',
+        'label,content,label,content',
+    ],
+)
+def test_set_secret(
+    secrets_context: Context[SecretsCharm], flow: str, lookup_by: Literal['id', 'label']
+):
+    secret = Secret({'some': 'content'}, owner='app', id='theid', label='thelabel')
+    state = State(leader=True, secrets={secret})
+    params = {'flow': flow}
+    match lookup_by:
+        case 'id':
+            params['secretid'] = 'theid'
+        case 'label':
+            params['secretlabel'] = 'thelabel'
+    state = secrets_context.run(secrets_context.on.action('set-secret-flow', params=params), state)
+    scenario_secret = next(iter(state.secrets))
```

Comment:
I think generally it's better to explicitly ask for the secret, rather than the "convert to Iterator and get first item" paradigm. Something like:

```suggestion
    scenario_secret = state.get_secret(id='theid') if lookup_by == 'id' else state.get_secret(label='thelabel')
```